openedx/core/djangoapps/course_groups/views.py

```diff
@@ -145,18 +153,59 @@ def course_cohort_settings_handler(request, course_key_string):
 
         settings_to_change = {}
 
-        if 'is_cohorted' in request.json:
-            settings_to_change['is_cohorted'] = request.json.get('is_cohorted')
-
-        if 'cohorted_course_wide_discussions' in request.json or 'cohorted_inline_discussions' in request.json:
+        if 'divided_course_wide_discussions' in request.json or 'divided_inline_discussions' in request.json:
             divided_course_wide_discussions = request.json.get(
-                'cohorted_course_wide_discussions', divided_course_wide_discussions
+                'divided_course_wide_discussions', divided_course_wide_discussions
             )
             divided_inline_discussions = request.json.get(
-                'cohorted_inline_discussions', divided_inline_discussions
+                'divided_inline_discussions', divided_inline_discussions
             )
             settings_to_change['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
 
+        if 'always_divide_inline_discussions' in request.json:
+            settings_to_change['always_divide_inline_discussions'] = request.json.get(
+                'always_divide_inline_discussions'
+            )
+
+        if not settings_to_change:
+            return JsonResponse({"error": unicode("Bad Request")}, 400)
+
+        try:
+            if settings_to_change:
+                discussion_settings = set_course_discussion_settings(course_key, **settings_to_change)
+
+        except ValueError as err:
+            # Note: error message not translated because it is not exposed to the user (UI prevents this state).
+            return JsonResponse({"error": unicode(err)}, 400)
+
+    return JsonResponse(_get_course_discussion_settings_representation(
+        course,
+        discussion_settings
+    ))
+
+
+@require_http_methods(("GET", "PATCH"))
+@ensure_csrf_cookie
+@expect_json
+@login_required
+def course_cohort_settings_handler(request, course_key_string):
+    """
+    The restful handler for cohort setting requests. Requires JSON.
+    This will raise 404 if user is not staff.
+    GET
+        Returns the JSON representation of cohort settings for the course.
+    PATCH
+        Updates the cohort settings for the course. Returns the JSON representation of updated settings.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    course = get_course_with_access(request.user, 'staff', course_key)
+
+    if request.method == 'PATCH':
+        settings_to_change = {}
+
+        if 'is_cohorted' in request.json:
+            settings_to_change['is_cohorted'] = request.json.get('is_cohorted')
+
         if 'always_cohort_inline_discussions' in request.json:
```

Comment:
You shouldn't be able to change these discussion division settings anymore in course_cohort_settings_handler. Please remove the code and any associated tests.